pkg/validation/validation.go

```diff
@@ -178,3 +197,16 @@ func getResourceById(id string, resources []core.CloudResource) core.ResourceKey
 	}
 	return resource
 }
+
+func validateNoDuplicateIds[T core.CloudResource](result *core.CompilationResult) error {
+	unitIds := make(map[string]struct{})
+	units := core.GetResourcesOfType[T](result)
+	for _, unit := range units {
+		fmt.Println(unit)
+		if id, ok := unitIds[unit.Key().Name]; ok {
+			return fmt.Errorf("Multiple Persist objects with the same name, '%s'", id)
```

Comment:
Won't `id` be a `struct{}{}`?

Reply:
ah yeah, meant to swithc that and log name. will update